evergreen_lint/config.py

```diff
@@ -140,6 +140,12 @@ def load_config(fh) -> Config:
     - rule: "variant-expansions"
       require_expansions: {[]}
       prohibit_expansions: {[]}
+
+    # Prevent experimental tasks from being used in non-experimental variants
+    - rule: "prevent-experimental-tasks-in-non-experimental-variants"
+      task_experimental_tag: "experimental"
+      variant_no_experimental_tag: "no_task_tag_experimental"
+      ignored_tasks: {[]}
```

Comment:
I think it's too specific. It would be better to have it more agnostic.

I'd propose this

```suggestion
    - rule: "prevent-tasks-with-tag-on-variant"
      variant_tag_prefix: "no_task_tag_"
      ignored_tasks: {[]}
```

Task tag to be banned will be the part after "no_task_tag_" prefix.

Or we can keep similar idea, but make the following change

```suggestion
    - rule: "prevent-tasks-with-tag-on-variant"
      tags:
        - variant_tag_name: "no_task_tag_experimental"
          prevent_task_tag: "experimental"
          ignored_tasks: {[]}
```

I expect to use this linting not just for "experimental" tag. And I think the current linter design doesn't allow to have multiple rules with the same name in the configuration.

Reply:
Thank you for your feedback. I have updated the PR to reflect your comments.